params.go

```diff
@@ -30,6 +30,58 @@ type Params struct {
 	Data interface{} `json:"-"`
 }
 
+// CommonParams should be used where there are no operation specific Params defined
+type CommonParams struct {
+	Params
+}
+
+func (list *CommonParams) toParams() *Params {
+	return &Params{
+		IdempotencyKey: list.IdempotencyKey,
+		Header:         list.Header,
+		Context:        list.Context,
+		RequestParams:  list,
+	}
+}
+
+// Option allows for customizations to a request.
+type Option func(p *Params)
+
+// CommonParamsWithOptions will create a CommonParams and apply the passed Options.
+func CommonParamsWithOptions(opts []Option) *CommonParams {
+	c := &CommonParams{}
+	c.Params.ApplyOptions(opts)
+	return c
+}
+
+// ApplyOptions will apply the passed Options to the Params.
+func (p *Params) ApplyOptions(opts []Option) {
```

Comment:
can we keep the signature the same here as in other calls where we take a variadic argument for opts instead of a slice? just for consistency